docs/reference/builder.md

```diff
@@ -954,6 +954,41 @@ For example:
 The output of the final `pwd` command in this `Dockerfile` would be
 `/path/$DIRNAME`
 
+## ARG
+    ARG [--default=<value>] [--description=<value>] <arg name>
+
+The `ARG` instruction defines a variable that is allowed to be passed as
+build-time variable to the builder. It is used to define a single variable at a time.
+Multiple variables may be defined by specifying `ARG` more than once. The `ARG`
+instruction can optionally take default value and description of the variable
+being defined as well. These are persisted in the image metadata so they can be
+useful for documentation. Additionally, the default value of the variable is used
+for expansion/substitution or passing to the `RUN` command when a build-time
+variable is not passed to the builder.
+
+A variable definition comes into effect from the line it appears in the `Dockerfile`
+i.e. prior to a build-time variable being defined by an ARG command any use of that 
+variable will result in an empty string. For example:
+
+    $ docker build --build-arg user=foo1 - <<EOF
+    USER ${user:-foo}
+    ARG $user
```

Comment:
thanks, I wasn't sure LOL

I will try to do a full review tonight, if time permits